packages/grafana-ui/src/components/Modal/Modal.tsx

```diff
@@ -78,16 +71,13 @@ export function Modal(props: PropsWithChildren<Props>) {
       <div
         className={styles.modalBackdrop}
         onClick={onClickBackdrop || (closeOnBackdropClick ? onDismiss : undefined)}
+        {...underlayProps}
       />
       <FocusScope contain={trapFocus} autoFocus restoreFocus>
-        {/*
-          tabIndex=-1 is needed here to support highlighting text within the modal when using FocusScope
-          see https://github.com/adobe/react-spectrum/issues/1604#issuecomment-781574668
-        */}
-        <div tabIndex={-1} className={cx(styles.modal, className)}>
+        <div className={cx(styles.modal, className)} ref={ref} {...overlayProps} {...dialogProps}>
           <div className={headerClass}>
-            {typeof title === 'string' && <DefaultModalHeader {...props} title={title} />}
-            {typeof title !== 'string' && title}
+            {typeof title === 'string' && <DefaultModalHeader {...props} title={title} id={titleProps.id} />}
+            {typeof title !== 'string' && <div id={titleProps.id}>{title}</div>}
```

Comment:
i'm not sure this is a viable solution as in practice any element can be put inside the title, so if passing a complex component to be rendered as a title, its text content won't necessarily label the modal itself. the alternative is not labelling the modal at all.

maybe the alternative here is requiring manual labelling when `title` is not a string?

any feedback would be highly appreciated

Reply:
another temporary solution could be to avoid labelling it for the time being and decide on wether we actually want modal dialogs without standardised titles, and possibly remove the ability to use custom components as titles, but this needs a broader discussion